beacon-chain/sync/validate_sync_contribution_proof.go

```diff
@@ -318,6 +330,67 @@ func (s *Service) setSyncContributionIndexSlotSeen(slot types.Slot, aggregatorIn
 	s.seenSyncContributionCache.Add(string(b), true)
 }
 
+// Set sync contribution's slot, root, committee index and bits.
+func (s *Service) setSyncContributionBits(c *ethpb.SyncCommitteeContribution) error {
+	s.syncContributionBitsOverlapLock.Lock()
+	defer s.syncContributionBitsOverlapLock.Unlock()
+	b := append(c.BlockRoot, bytesutil.Bytes32(uint64(c.Slot))...)
```

Comment:
I think you should copy `c.BlockRoot` first before appending it here. Due to how protobuf unmarshalling is carried out, you could unintentionally mutate this sync contribution.